server/src/session.ts

```diff
@@ -49,6 +49,13 @@ export class Session {
   private readonly logToConsole: boolean;
   private diagnosticsTimeout: NodeJS.Timeout|null = null;
   private isProjectLoading = false;
+  /**
+   * Tracks which `ts.server.Project`s have the renaming capability disabled.
+   *
+   * If we detect the compiler options diagnostic that suggests enabling strict mode, we want to
+   * disable renaming because we know that there are many cases where it will not work correctly.
+   */
+  private renameDisabledProjects: Set<ts.server.Project> = new Set();
```

Comment:
Should we use a `WeakSet` here so that projects that are closed can be gc-ed?

Reply:
👍  Done.